tests/TestCase/ORM/AssociationTest.php

```diff
@@ -330,6 +330,48 @@ public function testProperty()
     }
 
     /**
+     * Test that warning is shown if property name clashes with table field.
+     *
+     * @return void
+     * @expectedException PHPUnit_Framework_Error_Warning
```

Comment:
It would be better to add a notation `@expectedExceptionMessageRegExp /^Association property name "foo" clashes with field of same name of table "Foo"/`, to avoid to suppress warnings unintentionally.